### PR TITLE
[RW-2717][RW-3127] Convert attributes component to React

### PR DESCRIPTION
**Pull request by dolbeew**
- Convert `list-attributes` component to React, no design or functionality changes. (RW-2717)
- Fix issues when adding attributes in the Measurements wizard (RW-3127)
- Also includes updating the changelog to add categorical attributes to local data since they previously only existed in prod data

@freemabd to test functionality of UI
@jacmrob for React code

### CODE REVIEW

**Comment by dolbeew** [comment]
Nice catch, should be fixed now.

**Comment by dolbeew** [comment]
@jaycarlton I think I addressed all of your comments. Can you re-review when you have a chance?

File: ui/src/app/cohort-search/list-attributes-page/list-attributes-page.component.tsx [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+                if (!form.num.length) {
+                  form.num.push({
+                    name: AttrName.NUM,
+                    operator: null,
+                    operands: [],
+                    conceptId: node.conceptId,
+                    [attr.conceptName]: parseInt(attr.estCount, 10)
+                  });
+                } else {
+                  form.num[0][attr.conceptName] = parseInt(attr.estCount, 10);
```
[the remark this answers is not in the document]
Comment (jaycarlton) [reply ui/src/app/cohort-search/list-attributes-page/list-attributes-page.component.tsx]: OK, maybe it's eventually consistent or something.

File: ui/src/app/cohort-search/list-attributes-page/list-attributes-page.component.tsx [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+          acc += optionUtil[attr.operator].code;
+        }
+        return acc;
+      }, '');
+      return `param${(conceptId || id) + code}`;
+    }
+
+    get displayName() {
+      const {node: {name}} = this.props;
+      return stripHtml(name);
```
[the remark this answers is not in the document]
Comment (jaycarlton) [reply ui/src/app/cohort-search/list-attributes-page/list-attributes-page.component.tsx]: cool.

File: ui/src/app/cohort-search/list-attributes-page/list-attributes-page.component.tsx [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+                  form.cat.push(attr);
+                }
+              }
+            });
+            this.setState({form, loading: false});
+          });
+      } else {
+        options.unshift({label: 'Any', value: AttrName[AttrName.ANY]});
+        form.num = node.subtype === CriteriaSubType[CriteriaSubType.BP]
+          ? JSON.parse(JSON.stringify(PREDEFINED_ATTRIBUTES.BP_DETAIL))
```
[the remark this answers is not in the document]
Comment (jaycarlton) [reply ui/src/app/cohort-search/list-attributes-page/list-attributes-page.component.tsx]: Interesting. I wonder if there's a clone function or similar for that. Or just saving off the properties you need. I suppose this only happened once, but it's expensive to round-trip to string.

File: ui/src/app/cohort-search/list-attributes-page/list-attributes-page.component.tsx [hunk 4]
```diff
@@ -0,0 +1,553 @@
...
+      return this.props.node.subtype === CriteriaSubType.BP;
+    }
+
+    render() {
+      const {node} = this.props;
+      const {calculating, count, countError, form, loading, options} = this.state;
+      const {formErrors, formValid} = this.validateForm();
+      const disabled = calculating || form.exists || !formValid
+        || form.num.every(attr => attr.operator === 'ANY');
+      return (loading ? <SpinnerOverlay/> : <div style={{margin: '0.5rem 0 1.5rem'}}>
```
[the remark this answers is not in the document]
Comment (jaycarlton) [reply ui/src/app/cohort-search/list-attributes-page/list-attributes-page.component.tsx]: When I see `? type1 : type2` or something, it feels like a missed opportunity for polymorphism somewhere. I.e. there are invariants about items in each branch most likely that might not get enforced as strongly in this style.

**Comment by jaycarlton** [comment]
LGTM. I don't think I realized CB was a separate squad when I reviewed this. Thanks!

**Comment by dolbeew** [comment]
@jaycarlton We are separate squads but we do request/review PRs between teams regularly. Plus it's always good to get a fresh perspective from someone not as familiar with this part of the project. Thanks again!

File: ui/src/app/cohort-search/constant.ts [hunk 5]
```diff
@@ -143,16 +143,16 @@ export const PREDEFINED_ATTRIBUTES = {
     {
       conceptId: 903118,
       name: 'Systolic',
-      operands: [null],
-      operator: null,
+      operands: [],
+      operator: AttrName.ANY,
```
[the remark this answers is not in the document]
Comment (jaycarlton) [reply ui/src/app/cohort-search/constant.ts]: Sounds good. Sorry to leave this open. Is there a way to leave a non-blocking comment?

File: ui/src/app/cohort-search/list-attributes-page/list-attributes-page.component.tsx [hunk 6]
```diff
@@ -0,0 +1,553 @@
...
+          .then(resp => {
+            resp.items.forEach(attr => {
+              if (attr.type === AttrName[AttrName.NUM]) {
+                if (!form.num.length) {
+                  form.num.push({
+                    name: AttrName.NUM,
+                    operator: null,
+                    operands: [],
+                    conceptId: node.conceptId,
+                    [attr.conceptName]: parseInt(attr.estCount, 10)
```
[the remark this answers is not in the document]
Comment (jaycarlton) [reply ui/src/app/cohort-search/list-attributes-page/list-attributes-page.component.tsx]: OK. In general I don't expect to see numeric identifiers assigned to name fields. Especially in a language where type checking can't always be taken for granted. Likewise, even if you had a name that was a number (like Eleven in stranger things), I'm surprised to see a count on the RHS.

**Pull request merged**